### PR TITLE
99 implement version call

### LINKED ISSUE #99: Implement version call
skalenetwork/internal-support#27

### PR DESCRIPTION
**Pull request by yavrsky**

### CODE REVIEW

File: contracts/test/SafeMock.sol [hunk 1]
```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (DimaStebaev) [inline contracts/test/SafeMock.sol:1]: Why do you need to add `SafeMock` here?
It comes from `upgrade-tools`.

File: .solcover.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    skipFiles: ['interfaces/', 'test/']
```
Comment (DimaStebaev) [inline .solcover.js:2]: Why is skipping of `interfaces/` folder needed?

File: contracts/test/SafeMock.sol (hunk 1 above)
Comment (payvint) [inline contracts/test/SafeMock.sol:1]: Suggest to remove, it is already included in upgrade-tools

File: dictionaries/libraries.txt [hunk 3]
```diff
@@ -13,3 +13,5 @@ setuptools
 solhint
 typechain
 venv
+nomicfoundation
+
```
Comment (payvint) [inline dictionaries/libraries.txt:17]: Extra space

File: contracts/test/ImaMock.sol [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+/*
+    ImaMock.sol - SKALE Manager
+    Copyright (C) 2021-Present SKALE Labs
```
Comment (payvint) [inline contracts/test/ImaMock.sol]: Update info:)

File: contracts/test/MarionetteMock.sol [hunk 5]
```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+/*
+    Marionette.sol - SKALE Manager
```
Comment (payvint) [inline contracts/test/MarionetteMock.sol]: Name and info:)

**Review by payvint — DISMISSED** [review]

**Review by payvint — APPROVED** [review]

File: contracts/test/ImaMock.sol [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    You should have received a copy of the GNU Affero General Public License
+    along with SKALE Manager.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.8.11;
+
+import "@skalenetwork/ima-interfaces/IMessageReceiver.sol";
+
+interface IImaMock {
```
Comment (DimaStebaev) [inline contracts/test/ImaMock.sol:26]: Maybe it's better to create `IMessageSender` interface with `postOutgoingMessage` function in the `ima-interfaces` repo, to inherit `IMessageProxy` from it and make `ImaMock` a child of `IMessageSender`?

In current implementation there is no guaranty that `ImaMock` has the same interface as `MessageProxy` does.

**Pull request merged**